docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+        <column name="uuid" required="false" type="VARCHAR" size="255"/>
+        <column name="name" required="true" type="VARCHAR" size="255"/>
+        <column name="is_active" required="true" default="false" type="BOOLEAN"/>
+
+        <unique name="spy_configurable_bundle_template-uuid">
+            <unique-column name="uuid"/>
+        </unique>
+
+        <index name="index-spy_configurable_bundle_template-name">
+            <index-column name="name"/>
+        </index>
+
+        <behavior name="uuid">
+            <parameter name="key_columns" value="id_configurable_bundle_template"/>
+        </behavior>
+
+        <behavior name="timestampable"/>
+
+        <id-method-parameter value="spy_configurable_bundle_template_pk_seq"/>
+    </table>
+</database>
+
+```
+
+Injecting columns from `Product` to a `Url` `domain` by defining `spy_url.schema.xml` schema file in `Product` module.
+```xml
+<?xml version="1.0"?>
+<database xmlns="spryker:schema-01" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="zed" xsi:schemaLocation="spryker:schema-01 https://static.spryker.com/schema-01.xsd" namespace="Orm\Zed\Url\Persistence" package="src.Orm.Zed.Url.Persistence">
+
+    <table name="spy_url">
+        <column name="fk_resource_product_abstract" type="INTEGER"/>
+        <foreign-key name="spy_url-fk_resource_product_abstract" foreignTable="spy_product_abstract" phpName="SpyProduct" onDelete="CASCADE">
+            <reference foreign="id_product_abstract" local="fk_resource_product_abstract"/>
+        </foreign-key>
+    </table>
+</database>
+```
+
+## Provider / Router
+
+```
+[Organization]
+└── Yves
+    └── [Module]  
+        └── Plugin
+            ├── Provider
+            │   ├── [Name]ControllerProvider.php
+            │   └── [Name]ServiceProvider.php
+            └── [ConsumerModule]
+                └── [RouterName]Plugin.php
+```
+
+
+**Description**
+
+`Providers` and `Routers` are used during the bootstrap of [Yves](#applications) application. There are three types of them:
+- **Controller Provider** - Registers [Yves](#applications) [Controllers](#controller) of a module and binds them to a path.
+- **Service Provider** - Represents a cross-functional [Yves](#applications) service (eg: a validator).
+- **Router** - Resolves a path into a [Yves](#applications) [Controller and Action](#controller).
+
+**Conventions**
+- All `providers` and `routers` MUST be registered in `\[Organization]\Yves\ShopApplication\YvesBootstrap.php`.
+- A `Controller Provider` MUST extend `\SprykerShop\Yves\ShopApplication\Plugin\Provider\AbstractYvesControllerProvider`.
+- A `Service Provider` MUST implement `\Silex\ServiceProviderInterface and may extend \Spryker\Yves\Kernel\AbstractPlugin`.
+- A `Router` MUST extend `\SprykerShop\Yves\ShopRouter\Plugin\Router\AbstractRouter`.
+- A provider is classified as a [Plugin](#plugin) and so the plugin's conventions apply.
+
+## Query Container facade
```

Comment:
no we can't, because it's not deprecated. we are still supporting it, and rolling out fixes for it. yet, we recommend it to be not implemented anymore, as this design has many flaws.

Reply:
but i've added an extra line, to use EM and Repository instead